vizro-core/src/vizro/models/_components/ag_grid.py

```diff
@@ -114,15 +114,21 @@ def _action_inputs(self) -> dict[str, _IdProperty]:
             **{ag_grid_prop: f"{self._inner_component_id}.{ag_grid_prop}" for ag_grid_prop in DAG_AG_GRID_PROPERTIES},
         }
 
-    def _get_value_from_trigger(self, value: str, trigger: list[dict[str, str]]) -> JsonValue:
-        """Value is the name of the column. There is only one row selected, so we just look at trigger[0]."""
-        # In case the selectedRows is empty (e.g. when the user unselects the row), trigger is an empty list.
-        # This works for both multi=True and multi=False selectors.
+    def _get_value_from_trigger(self, value: str, trigger: list[dict[str, str]]) -> MultiValueType | None:
+        """Extract values from the trigger that represents selected dag.AgGrid rows. Value is the name of the column.
+
+        Returns:
+          - list of values (one per point) or None if no points selected (signals reset).
+
+        Raises:
+          - ValueError if `value` column name can't be found.
+        """
+        # Returning None signals a reset of control to its original value.
         if not trigger:
-            return []
+            return None
 
         try:
-            return trigger[0][value]
+            return sorted({row[value] for row in trigger})
```

Comment:
Do we actually want to do `sorted`? I think probably not. Probably best to just send values in the same order that they are selected.